### DISCUSSION TITLE
Question about Edge Feature

### DISCUSSION THREAD

**Discussion by mickeyhan** [pyg-team/pytorch_geometric#4490] [Q&A]
Hi, Thank you very much for your time in reading my question . newbie here sorry if I'm not making sense
 
<img width="300" alt="question" src="https://user-images.githubusercontent.com/82403116/163710826-35a6743e-88cc-41bc-a8e2-78b481bfd71f.PNG">

if I have a bunch of directed graphs with varying number of nodes and edges, and individually I create some subgraphs and want edge feature in this case [0,3] to be concat of the edges in between its start and end and pass through a nn.Linear() layer, if the number of nodes and edges vary from cases to cases, some are five, some are ten or any number etc. how should I approach, by padding them all to the same size ? , or engineers using graphs dont do things like this ?

Thank you very much for your time in reading my question.

**Comment by rusty1s** [comment]
How about you use PyG `transforms` for this:
```python
def add_edge(data):
    loop = torch.tensor([[0], [data.num_nodes - 1]])
    data.edge_index = torch.cat([data.edge_index, loop], dim=1)
    return data

dataset = Dataset(…, transform=add_edge)
```

**Reply by mickeyhan** [reply]
Thank you very much for your reply ! Have a great day !
